cmake/modules/FindGnuLd.cmake

```diff
@@ -29,7 +29,7 @@ execute_process(COMMAND ${CMAKE_C_COMPILER} --print-prog-name=ld.bfd
                 OUTPUT_STRIP_TRAILING_WHITESPACE)
 
 if(EXISTS "${GNULD_LINKER}")
-  set(GNULD_LINKER_IS_BFD TRUE)
+  set(GNULD_LINKER_IS_BFD ON CACHE BOOL "Is linker ld.bfd compatible")
```

Comment:
The lookup at those locations has already set the cache variable:
https://github.com/zephyrproject-rtos/zephyr/blob/bbc4f11fb3a35d84568c53340a6a934777f71461/cmake/modules/FindGnuLd.cmake#L46
https://github.com/zephyrproject-rtos/zephyr/blob/bbc4f11fb3a35d84568c53340a6a934777f71461/cmake/modules/FindGnuLd.cmake#L49

so only chance that `GNULD_LINKER_IS_BFD` has already been set, but the corresponding `GNULD_LINKER_IS_BFD` is undefined is when a user has specifically pointed to a ld through the use of: `-DGNULD_LINKER_IS_BFD=<forced-linker-to-use>`.

In such a case we cannot really know if linker is bfd compatible or not, as we have no proper detection mechanism.

So in the scenario described we must rely on user to provide us the correct setting, so I propose issue a warning in such case.

Something like:
```
if(EXISTS "${GNULD_LINKER}")
  if(NOT DEFINED GNULD_LINKER_IS_BFD)
    message(WARNING "GNULD_LINKER specified directly in cache, but GNULD_LINKER_IS_BFD is not defined."
                    "Assuming GNULD_LINKER_IS_BFD as OFF, please set GNULD_LINKER_IS_BFD to correct value in cache to silence this warning"
    )
    set(GNULD_LINKER_IS_BFD OFF)
  endif()
endif()
```

Reply:
Added the warning and also to return early if variable is already set.